framework_tvb/tvb/tests/framework/core/services/project_service_test.py

```diff
@@ -469,93 +456,96 @@ def test_update_meta_data_simple(self):
         op_meta = XMLReader(op_path).read_metadata()
         assert op_meta['user_group'] == 'new user group', 'UserGroup not updated!'
 
-
-    def test_update_meta_data_group(self):
+    def test_update_meta_data_group(self, datatype_group_factory):
         """
         Test the new update metaData for a group of dataTypes.
         """
-        datatypes, group_id = TestFactory.create_group(self.test_user, subject="test-subject-1")
+        group = datatype_group_factory()
 
         new_meta_data = {DataTypeOverlayDetails.DATA_SUBJECT: "new subject",
                          DataTypeOverlayDetails.DATA_STATE: "updated_state",
-                         DataTypeOverlayDetails.CODE_OPERATION_GROUP_ID: group_id,
+                         DataTypeOverlayDetails.CODE_OPERATION_GROUP_ID: group.id,
                          DataTypeOverlayDetails.CODE_OPERATION_TAG: 'newGroupName'}
-        self.project_service.update_metadata(new_meta_data)  
-          
+        self.project_service.update_metadata(new_meta_data)
+        datatypes = dao.get_datatype_in_group(group.id)
         for datatype in datatypes:
             new_datatype = dao.get_datatype_by_id(datatype.id)
-            assert group_id == new_datatype.parent_operation.fk_operation_group
-            new_group = dao.get_generic_entity(model.OperationGroup, group_id)[0]
+            assert group.id == new_datatype.parent_operation.fk_operation_group
+            new_group = dao.get_generic_entity(model_operation.OperationGroup, group.id)[0]
             assert new_group.name == "newGroupName"
             self.__check_meta_data(new_meta_data, new_datatype)
-            
-    
+
     def _create_datatypes(self, dt_factory, nr_of_dts):
         for idx in range(nr_of_dts):
             dt = Datatype1()
             dt.row1 = "value%i" % (idx,)
             dt.row2 = "value%i" % (idx + 1,)
             dt_factory._store_datatype(dt)
-            
-            
-    def test_retrieve_project_full(self):
+
+    def test_retrieve_project_full(self, dummy_datatype_index_factory, dummy_datatype_index_factory2, dummy_datatype_index_factory3):
         """
         Tests full project information is retrieved by method `ProjectService.retrieve_project_full(...)`
         """
-        dt_factory = datatypes_factory.DatatypesFactory()
-        self._create_datatypes(dt_factory, 3)
-        _, ops_nr, operations, pages_no = self.project_service.retrieve_project_full(dt_factory.project.id)
+
+        project = TestFactory.create_project(self.test_user)
+        operation = TestFactory.create_operation(test_user=self.test_user, test_project=project)
+
+        dt1 = dummy_datatype_index_factory(project=project, operation=operation)
+        dt2 = dummy_datatype_index_factory2(project=project, operation=operation)
+        dt3 = dummy_datatype_index_factory3(project=project, operation=operation)
```

Comment:
I see here the strange factory2/3 used. Why can't we simply call the same factory 3 times?